core/src/test/java/org/elasticsearch/common/xcontent/builder/XContentBuilderTests.java

```diff
@@ -260,4 +263,36 @@ public void testCopyCurrentStructure() throws Exception {
 
         assertThat(i, equalTo(terms.size()));
     }
+
+    @Test
+    public void testHandlingOfPath() throws IOException {
+        Path path = PathUtils.get("path");
```

Comment:
Ok, adding a test like that is easy enough, even though I still feel it is redundant.
It also created enough repetition that I felt we should extract the common check code.